modules/core/src/convert.cpp

```diff
@@ -5371,6 +5371,52 @@ static bool ocl_convertScaleAbs( InputArray _src, OutputArray _dst, double alpha
     return k.run(2, globalsize, NULL, false);
 }
 
+static bool ocl_convertFp16( InputArray _src, OutputArray _dst )
+{
+    int type = _src.type(), depth = CV_MAT_DEPTH(type), cn = CV_MAT_CN(type);
+
+    int nSChannels = _src.getMat().channels();
```

Comment:
`.getMat()` is expensive operation (here and below) - it maps (or downloads) OpenCL buffer into CPU memory.
There is `cn` variable above for this value.

`_dst.create()` -> `_dst.createSameSize()`, so dims is not needed.